src/emailagent.cpp

```diff
@@ -669,24 +669,31 @@ void EmailAgent::expungeMessages(const QMailMessageIdList &ids)
     exportUpdates(accountList);
 }
 
-void EmailAgent::downloadAttachment(int messageId, const QString &attachmentlocation)
+void EmailAgent::downloadAttachment(int messageId, const QString &attachmentLocation)
 {
     m_messageId = QMailMessageId(messageId);
     QMailMessage message (m_messageId);
-
+    QMailMessagePart attachmentPart;
+    bool partFound = false;
```

Comment:
It could work, but I had troubles verifying the validity of the returned part IIRC, looking now at the code there's two function one that returns a Const QMailMessagePart and asserts if the part is not found(not good here) and a second one that just returns a memory address to a internally create part: QMailMessagePart* part = 0, so I don't see a clear way to verify this, since also didn't find a value that all parts must have.